src/libraries/System.Private.CoreLib/src/System/Numerics/BitOperations.cs

```diff
@@ -708,5 +711,168 @@ public static nuint RotateRight(nuint value, int offset)
             return (nuint)RotateRight((uint)value, offset);
 #endif
         }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * uint32_t __crc32b (uint32_t a, uint8_t b)
+        /// * uint32_t __crc32cb (uint32_t a, uint8_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, byte data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // uint32_t __crc32b (uint32_t a, uint8_t b)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32cb (uint32_t a, uint8_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+            int tableIndex = (int)((crc ^ data) & 0xFF);
+            crc = s_crcTable[tableIndex] ^ (crc >> 8);
+
+            return crc;
+        }
+
+        /// <summary>
+        /// Calculates the CRC (Cyclic redundancy check) checksum.
+        ///
+        /// Uses following Hardware Intrinsics, if supported:
+        /// * unsigned int _mm_crc32_u16 (unsigned int crc, unsigned short v)
+        /// * uint32_t __crc32ch (uint32_t a, uint16_t b)
+        /// </summary>
+        /// <param name="crc">The base value to calculate checksum on</param>
+        /// <param name="data">The checksum data</param>
+        /// <returns>The CRC-checksum</returns>
+        [CLSCompliant(false)]
+        public static uint Crc32C(uint crc, ushort data)
+        {
+            if (Sse42.IsSupported)
+            {
+                // unsigned int _mm_crc32_u16 (unsigned int crc, unsigned short v)
+                return Sse42.Crc32(crc, data);
+            }
+            if (Crc32.IsSupported)
+            {
+                // uint32_t __crc32ch (uint32_t a, uint16_t b)
+                return Crc32.ComputeCrc32C(crc, data);
+            }
+
+
+            // Software fallback
+            if (!BitConverter.IsLittleEndian)
+            {
+                data = BinaryPrimitives.ReverseEndianness(data);
+            }
+
+            ref uint lut = ref MemoryMarshal.GetArrayDataReference(s_crcTable);
+
+            crc = Unsafe.Add(ref lut, (nint)(byte)(crc ^ (byte)data)) ^ (crc >> 8);
+            data >>= 8;
+            crc = Unsafe.Add(ref lut, (nint)(byte)(crc ^ (byte)data)) ^ (crc >> 8);
```

Comment:
```suggestion
            crc = Unsafe.Add(ref lut, (nint)(byte)(crc ^ data)) ^ (crc >> 8);
```

Cast here is unnecessary because the the upper byte was cleared by the shift above.